src/plugin.js

```diff
@@ -0,0 +1,314 @@
+      // the overlay has shown since the last seek. This means that we need
+      // to be sure we aren't re-showing it at a later time than it is
+      // scheduled to appear.
+      return Math.floor(time) === start;
+    }
+
+    return start === type;
+  }
+
+  /**
+   * Event listener that can trigger the overlay to show.
+   *
+   * @param  {Event} e
+   */
+  startListener_(e) {
+    let time = this.player().currentTime();
+
+    if (this.shouldShow_(time, e.type)) {
+      this.show();
+    }
+  }
+
+  /**
+   * Event listener that can trigger the overlay to show.
+   *
+   * @param  {Event} e
+   */
+  endListener_(e) {
+    let time = this.player().currentTime();
+
+    if (this.shouldHide_(time, e.type)) {
+      this.hide();
+    }
+  }
+
+  /**
+   * Event listener that can looks for rewinds - that is, backward seeks
+   * and may hide the overlay as needed.
+   *
+   * @param  {Event} e
+   */
+  rewindListener_(e) {
+    let time = this.player().currentTime();
+    let previous = this.previousTime_;
+    let start = this.options_.start;
+    let end = this.options_.end;
+
+    // Did we seek backward?
+    if (time < previous) {
+      this.debug('rewind detected');
+
+      // The overlay remains visible if two conditions are met: the end value
+      // MUST be an integer and the the current time indicates that the
+      // overlay should NOT be visible.
+      if (isInteger(end) && !this.shouldShow_(time)) {
+        this.debug(tsmlj`
+          hiding; ${end} is an integer and overlay should not show at this time
+        `);
+        this.hasShownSinceSeek_ = false;
+        this.hide();
+
+      // If the end value is an event name, we cannot reliably decide if the
+      // overlay should still be displayed based solely on time; so, we can
+      // only queue it up for showing if the seek took us to a point before
+      // the start time.
+      } else if (hasNoWhitespace(end) && time < start) {
+        this.debug(tsmlj`
+          hiding; show point (${start}) is before now (${time}) and end
+          point (${end}) is an event
+        `);
+        this.hasShownSinceSeek_ = false;
+        this.hide();
+      }
+    }
+
+    this.previousTime_ = time;
+  }
+}
+
+videojs.registerComponent('Overlay', Overlay);
+
+/**
+ * Initialize the plugin.
+ *
+ * @function plugin
+ * @param    {Object} [options={}]
+ */
+const plugin = function(options) {
+  const settings = videojs.mergeOptions(defaults, options);
+
+  // De-initialize the plugin if it already has an array of overlays.
+  if (Array.isArray(this.overlays_)) {
+    this.overlays_.forEach(overlay => overlay.dispose());
+  }
+
+  const overlays = settings.overlays;
+
+  // We don't want to keep the original array of overlay options around
+  // because it doesn't make sense to pass it to each Overlay component.
+  delete settings.overlays;
+
+  this.overlays_ = overlays.map(o => {
+    this.addChild('Overlay', videojs.mergeOptions(settings, o));
```

Comment:
Well, the `{}` change the meaning of the function. I guess it works because `addChild` just upper cases the first letter.